core-build/gulp-core-build-sass/src/SassTask.ts

```diff
@@ -76,218 +107,214 @@ export class SassTask extends GulpTask<ISassTaskConfig> {
   }
 
   public loadSchema(): Object {
-    return require('./sass.schema.json');
+    return JsonFile.load(path.join(__dirname, 'sass.schema.json'));
   }
 
-  public executeTask(
-    gulp: typeof Gulp,
-    completeCallback: (error?: string) => void
-  ): Promise<{}> | NodeJS.ReadWriteStream | void {
+  public executeTask(gulp: typeof Gulp, completeCallback: (error?: string) => void): Promise<void> | undefined {
 
     if (!this.taskConfig.sassMatch) {
       completeCallback('taskConfig.sassMatch must be defined');
       return;
     }
 
-    /* tslint:disable:typedef */
-    const autoprefixer = require('autoprefixer');
-    const cssModules = require('postcss-modules');
-    /* tslint:enable:typedef */
+    return this._globAll(...this.taskConfig.sassMatch).then((matches: string[]) => {
+      return Promise.all(matches.map((match) => this._processFile(match)));
```

Comment:
>Promise [](start = 13, length = 7)

I feel like we should either be using `Promise` or `completeCallback`, but not both.

IMO the API itself was designed wrong -- it should have had separate signatures for the two usage patterns.